### PR TITLE
NMS-7553: Add Juniper SRX flow performance monitoring and default thresholds

### PR DESCRIPTION
**Pull request by StCyr**
Added configuration to collect, graph and threshold on Juniper SRX router's jnxJsSPUMonitoringCurrentFlowSession OID.

Background: 

When the number of tracked sessions reach the router's maximu capacity, the router starts to drop arbitrary sessions (not necessarly inactive ones) when needed to stay below the maximum. This, obviously, creates strange network issues.

This can easily happen if you forget to define a timeout for some kind of UDP sessions.

JIRA: http://issues.opennms.org/browse/NMS-7553
Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-NMS258

Todo:
- [x] Review
- [x] Pass unit tests
- [ ] Merge to develop

### CODE REVIEW

File: opennms-base-assembly/src/main/filtered/etc/datacollection/juniper.xml [hunk 1]
```diff
@@ -44,6 +52,12 @@
         <mibObj oid=".1.3.6.1.4.1.4874.2.2.2.1.9.4.1.4" instance="juniSystemTempIndex" alias="juniSTPhysicalIndex" type="string" />
       </group>
 
+      <group name="juniper-srx-router" ifType="all">
+        <mibObj oid=".1.3.6.1.4.1.2636.3.39.1.12.1.1.1.11" instance="jnxJsSPUMonitoringObjectsTable" alias="jnxJsSPUMonitoringNodeDescr"          type="string" />
+        <mibObj oid=".1.3.6.1.4.1.2636.3.39.1.12.1.1.1.6"  instance="jnxJsSPUMonitoringObjectsTable" alias="jnxJsSPUMonitoringCurrentFlowSession" type="Gauge32" />
+        <mibObj oid=".1.3.6.1.4.1.2636.3.39.1.12.1.1.1.7"  instance="jnxJsSPUMonitoringObjectsTable" alias="jnxJsSPUMonitoringMaxFlowSession"     type="Gauge32" />
```
Comment (indigo423) [inline opennms-base-assembly/src/main/filtered/etc/datacollection/juniper.xml]: The `alias` exceeds 19 character which is the maximum in RRD world.

File: opennms-base-assembly/src/main/filtered/etc/datacollection/juniper.xml [hunk 2]
```diff
@@ -44,6 +52,12 @@
         <mibObj oid=".1.3.6.1.4.1.4874.2.2.2.1.9.4.1.4" instance="juniSystemTempIndex" alias="juniSTPhysicalIndex" type="string" />
       </group>
 
+      <group name="juniper-srx-router" ifType="all">
+        <mibObj oid=".1.3.6.1.4.1.2636.3.39.1.12.1.1.1.11" instance="jnxJsSPUMonitoringObjectsTable" alias="jnxJsSPUMonitoringNodeDescr"          type="string" />
+        <mibObj oid=".1.3.6.1.4.1.2636.3.39.1.12.1.1.1.6"  instance="jnxJsSPUMonitoringObjectsTable" alias="jnxJsSPUMonitoringCurrentFlowSession" type="Gauge32" />
```
Comment (indigo423) [inline opennms-base-assembly/src/main/filtered/etc/datacollection/juniper.xml]: The alias exceeds 19 character which is the maximum in RRD world.

File: opennms-base-assembly/src/main/filtered/etc/datacollection/juniper.xml [hunk 3]
```diff
@@ -44,6 +52,12 @@
         <mibObj oid=".1.3.6.1.4.1.4874.2.2.2.1.9.4.1.4" instance="juniSystemTempIndex" alias="juniSTPhysicalIndex" type="string" />
       </group>
 
+      <group name="juniper-srx-router" ifType="all">
+        <mibObj oid=".1.3.6.1.4.1.2636.3.39.1.12.1.1.1.11" instance="jnxJsSPUMonitoringObjectsTable" alias="jnxJsSPUMonitoringNodeDescr"          type="string" />
```
Comment (indigo423) [inline opennms-base-assembly/src/main/filtered/etc/datacollection/juniper.xml]: The alias exceeds 19 character which is the maximum in RRD world.

File: opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/juniper-graph.properties [hunk 4]
```diff
@@ -94,6 +95,29 @@ report.juniper.temp.command=--title="Current Temperature" \
...
+##### Juniper SRX reports
+#####
+
+report.srx.spuFlowMonitor.name=Juniper SRX Flows
+report.srx.spuFlowMonitor.columns=jnxJsSPUMonitoringCurrentFlowSession,jnxJsSPUMonitoringMaxFlowSession
+report.srx.spuFlowMonitor.type=jnxJsSPUMonitoringObjectsTable
+report.srx.spuFlowMonitor.propertiesValues=jnxJsSPUMonitoringNodeDescr
+report.srx.spuFlowMonitor.command=--title="Sessions used" \
+ --vertical-label="Sessions" \
+ DEF:currentSessions={rrd1}:jnxJsSPUMonitoringC:AVERAGE \
```
Comment (indigo423) [inline opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/juniper-graph.properties]: I think the cutted is the reason for the jnxJsSPUMonitoringCurrentFlowSession --> jnxJsSPUMonitoringC is a result of the exceeded alias name. I think refactoring to similar name make it easier.

File: opennms-base-assembly/src/main/filtered/etc/datacollection/juniper.xml (hunk 3 above)
Comment (agalue) [inline opennms-base-assembly/src/main/filtered/etc/datacollection/juniper.xml]: The restriction of the 19 characters for the DS name is for numeric variables only (i.e. integer, counter, gauge, etc. or whatever numeric metric that must be stored inside the RRD file). The MibObj defined as string can be as long as you wish.

File: opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/juniper-graph.properties [hunk 5]
```diff
@@ -94,6 +95,29 @@ report.juniper.temp.command=--title="Current Temperature" \
  GPRINT:val3:MAX:"Max  \\: %8.2lf %s\\n"
 
 #####
+##### Juniper SRX reports
+#####
+
+report.srx.spuFlowMonitor.name=Juniper SRX Flows
+report.srx.spuFlowMonitor.columns=jnxJsSPUMonitoringCurrentFlowSession,jnxJsSPUMonitoringMaxFlowSession
+report.srx.spuFlowMonitor.type=jnxJsSPUMonitoringObjectsTable
+report.srx.spuFlowMonitor.propertiesValues=jnxJsSPUMonitoringNodeDescr
```
Comment (indigo423) [inline opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/juniper-graph.properties]: Suggestion: this properties value is not used? Maybe it make sense in the graph title?

File: opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/juniper-graph.properties (hunk 4 above)
Comment (StCyr) [inline opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/juniper-graph.properties]: yep, tha's right: Inside the jrb file the datasource's name is trimmed to 19 characters :-)

I can fix that in my PR, no problem.

File: opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/juniper-graph.properties (hunk 5 above)
Comment (StCyr) [inline opennms-base-assembly/src/main/filtered/etc/snmp-graph.properties.d/juniper-graph.properties]: good suggestion

**Comment by StCyr** [comment]
Hi @indigo423 ,

I fixed the alias length.

that should be oK now.

**Comment by indigo423** [comment]
With your changes I've added a branch to see if the Unit tests pass and can fix those before we merge to develop. I've added the link to the Bamboo build for this issue. If this one goes green looks good to me to merge to develop.

Thanks for this nice enhancement.

**Pull request merged**